### PR TITLE
Convert hosted-cluster-config-operator into subcommand of CPO

### PR DESCRIPTION
**Pull request by ironcladlou**
This commit removes the discrete hosted-cluster-config-operator (HCCO) binary
and moves it to become a subcommand of control-plane-operator (CPO).

Architectural discussions have led to the conclusion that the logic of HCCO can
be incorporated into the CPO itself, since the programs are developed and
versioned in lockstep, coexist in the control plane namespace, and have very
similar scopes of access and responsibility. Eventually the HCCO could be part
of the CPO process itself, but this commit takes an incremental step towards
that end by relocating the code and removing the HCCO image.

There are two immediate benefits:

1. The control plane image that we incorporate into the OCP payload will become
   smaller (by shedding the 75MB HCCO binary).
2. The HCCO code will be closer to its final destination (a subpackage of CPO).

A future commit can merge HCCO logic into the CPO process and then remove all the
associated workload management code for HCCO (SA, role, deployment, etc.).

### CODE REVIEW

**Review by alvaroaleman — APPROVED** [review]

**Comment by sjenning** [comment]
I think this is legit failing CI
/hold
Does ^ block retries by openshift-bot?

**Comment by alvaroaleman** [comment]
> Does ^ block retries by openshift-bot?

Yes it does.

I think the failure was legit but fixed by the latest push.

**Comment by ironcladlou** [comment]
Yes, the failure was legit and fixed this morning

**Comment by sjenning** [comment]
Ah ok, didn't see the push
/hold cancel

**Review by csrwng — APPROVED** [review]

File: test/e2e/util/util.go [hunk 1]
```diff
@@ -79,8 +79,16 @@ func createCluster(t *testing.T, ctx context.Context, client crclient.Client, cl
 			t.Logf("Skipping cluster dump because no artifact dir was provided")
 		}
 		// TODO: Figure out why this is slow
-		//e2eutil.DumpGuestCluster(context.Background(), client, hostedCluster, globalOpts.ArtifactDir)
-		DestroyHostedCluster(t, ctx, hc, clusterMgr, artifactDir)
+		// e2eutil.DumpGuestCluster(context.Background(), client, hostedCluster, globalOpts.ArtifactDir)
+		destroyAttempt := 1
```
Comment (alvaroaleman) [inline test/e2e/util/util.go:83]: What is the purpose of this variable? There isn't any retrying, it is just logged once, then incremented, then unused

Comment (ironcladlou) [reply test/e2e/util/util.go:83]: DestroyHostedCluster retries internally via PollImmediateUntil

**Pull request merged**